### PR TITLE
Add network policies for WebLogic operator

### PR DESCRIPTION
**Pull request by bobdonat**
# Description

This pull request adds a network policy for the WebLogic operator and Coherence operator.  Each policy denys-all for ingress and allows an egress to the k8s api server.

Fixes VZ-2478 and VZ-2479

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/helm_config/charts/verrazzano/templates/08-thirdparty-networkpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  policyTypes:
+    - Ingress
+    - Egress
+  egress:
+    - ports:
+        - port: {{ .Values.kubernetes.service.endpoint.port }}
+          protocol: TCP
+      to:
+        - ipBlock:
+            cidr: {{ .Values.kubernetes.service.endpoint.ip }}/32
```
Comment (pfmackin) [inline platform-operator/helm_config/charts/verrazzano/templates/08-thirdparty-networkpolicy.yaml:25]: are we sure the /32 is always the correct value?   What if the pod is not on the master host (where the Kubernetes API server is)

File: platform-operator/scripts/install/3-install-verrazzano.sh [hunk 2]
```diff
@@ -8,8 +8,6 @@ SCRIPT_DIR=$(cd $(dirname "$0"); pwd -P)
 . $SCRIPT_DIR/config.sh
 
 CONFIG_DIR=$SCRIPT_DIR/config
-TMP_DIR=$(mktemp -d)
```
Comment (pfmackin) [inline platform-operator/scripts/install/3-install-verrazzano.sh:11]: Why were these 2 lines removed?

Comment (bobdonat) [reply platform-operator/scripts/install/3-install-verrazzano.sh:11]: TMP_DIR is not used in this script

File: platform-operator/helm_config/charts/verrazzano/templates/08-thirdparty-networkpolicy.yaml (hunk 1 above)
Comment (bobdonat) [reply platform-operator/helm_config/charts/verrazzano/templates/08-thirdparty-networkpolicy.yaml:25]: I am not sure. The examples I saw used 32.  Any idea how I would know if 32 is always the correct value?

Comment (bobdonat) [reply platform-operator/helm_config/charts/verrazzano/templates/08-thirdparty-networkpolicy.yaml:25]: the /32 allows for a single address as I understand.  I would not expect the K8s API endpoint to be a range of addresses.

Comment (bobdonat) [reply platform-operator/helm_config/charts/verrazzano/templates/08-thirdparty-networkpolicy.yaml:25]: OCI console shows the OKE cluster with: **Kubernetes API Private Endpoint: 10.0.0.9:6443**.  Same value we are using in the network policy

**Review by pfmackin — APPROVED** [review]

**Review by pfmackin — APPROVED** [review]

File: platform-operator/helm_config/charts/verrazzano/templates/08-thirdparty-networkpolicy.yaml (hunk 1 above)
Comment (wmhopkins) [reply platform-operator/helm_config/charts/verrazzano/templates/08-thirdparty-networkpolicy.yaml:25]: Should we allow for the /<bits> to be provided in the Values file?

File: platform-operator/helm_config/charts/verrazzano/templates/08-thirdparty-networkpolicy.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2021, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+---
+# Network policy for WebLogic operator
+# Ingress:  deny all
```
[the remark this answers is not in the document]
Comment (bobdonat) [reply platform-operator/helm_config/charts/verrazzano/templates/08-thirdparty-networkpolicy.yaml:5]: I will fix the spaces after ':' in the next PR.
Currently, there are no webhooks for WebLogic operator.

File: platform-operator/helm_config/charts/verrazzano/templates/08-thirdparty-networkpolicy.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+kind: NetworkPolicy
+metadata:
+  name: weblogic-operator
+  namespace: {{ .Release.Namespace }}
+spec:
+  podSelector:
+    matchLabels:
+      app: weblogic-operator
+  policyTypes:
+    - Ingress
```
[the remark this answers is not in the document]
Comment (bobdonat) [reply platform-operator/helm_config/charts/verrazzano/templates/08-thirdparty-networkpolicy.yaml:17]: Yes, this is how you specify deny-all for an ingress

**Comment by bobdonat** [comment]
>>>Should we allow for the / to be provided in the Values file?

Not sure how this would be used if we allowed the / in the Values file

**Pull request merged**